### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by phemmylintry**
@KeelTech

### CODE REVIEW

**Comment by phemmylintry** [comment]
@KeelTech

File: keel/api/v1/auth/views.py [hunk 1]
```diff
@@ -97,22 +90,125 @@ def login(self, request, format="json"):
...
+            "status" : 1,
+            "message" : ""
+        }
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.validated_data
+        reset_token = validated_data.get('reset_token', None)
+        password = validated_data['password']
+        try:
+            token = PasswordResetToken.objects.get(reset_token=reset_token)
```
Comment (KeelTech) [inline keel/api/v1/auth/views.py:159]: we also need to check if the token is not ore then 10 minutes old!!!!!...Also validate that this token belongs to the iuser it was sent to

File: keel/api/v1/auth/views.py [hunk 2]
```diff
@@ -97,22 +90,125 @@ def login(self, request, format="json"):
...
+        return Response(response)
+
+
+class ChangePasswordView(GenericViewSet):
+
+    serializer_class = serializers.ChangePasswordSerializer
+    permission_classes = (IsAuthenticated, )
+    authentication_classes = (JWTAuthentication, )
+
+    def change_password_without_email(self, request):
```
Comment (KeelTech) [inline keel/api/v1/auth/views.py:182]: whats this method for?

File: keel/api/v1/auth/views.py [hunk 3]
```diff
@@ -97,22 +90,125 @@ def login(self, request, format="json"):
...
+            token = PasswordResetToken.objects.get(reset_token=reset_token)
+            # get user and set password
+            user = User.objects.get(email=token.user)
+            user.set_password(password)
+            user.save()
+        except Exception as e:
+            logger.error('ERROR: AUTHENTICATION:ConfirmPasswordReset ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
```
Comment (KeelTech) [inline keel/api/v1/auth/views.py:168]: dont simploy throw a 400...
what if the db request failed
check for the type of exception and throw accordingly

File: keel/api/v1/auth/views.py (hunk 2 above)
Comment (phemmylintry) [reply keel/api/v1/auth/views.py:182]: an authenticated user can change their password

File: keel/api/v1/auth/views.py [hunk 4]
```diff
@@ -162,21 +258,46 @@ def post(self, request, *args, **kwargs):
...
+        }
+        token = request.headers.get('Authorization', None).split(" ")[1]
+
+        try:
+            CustomToken.objects.get(token=token).delete()
+        except Exception as e:
+            logger.error('ERROR: AUTHENTICATION:UserDeleteTokenView ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
```
Comment (KeelTech) [inline keel/api/v1/auth/views.py:294]: again , cant be a straight 400

File: keel/api/v1/auth/views.py (hunk 3 above)
Comment (phemmylintry) [reply keel/api/v1/auth/views.py:168]: Should I create separate try, except?

**Review by KeelTech — APPROVED** [review]

**Pull request merged**